src/commcare_cloud/commands/ansible/service.py

```diff
@@ -1,328 +1,444 @@
-        'proxy': ['nginx'],
-        'riakcs': ['riak', 'riak-cs', 'stanchion'],
-        'stanchion': ['stanchion'],
-        'es': ['elasticsearch'],
-        'redis': ['redis'],
-        'couchdb2': ['couchdb2'],
-        'postgresql': ['postgresql', 'pgbouncer'],
-        'rabbitmq': ['rabbitmq'],
-        'kafka': ['kafka', 'zookeeper'],
-        'pg_standby': ['postgresql', 'pgbouncer'],
-        'celery': ['celery'],
-        'webworkers': ['webworkers'],
-        'formplayer': ['formplayer-spring'],
-        'touchforms': ['formplayer'],
-    }
-    ACTIONS = ['start', 'stop', 'restart', 'status']
-    DESIRED_STATE_FOR_ACTION = {
-        'start': 'started',
-        'stop': 'stopped',
-        'restart': 'restarted',
-    }
-    # add this mapping where service group is not same as the inventory group itself
-    INVENTORY_GROUP_FOR_SERVICE = {
-        'stanchion': 'stanchion',
-        'elasticsearch': 'elasticsearch',
-        'zookeeper': 'zookeeper',
-    }
-    # add this if the service package is not same as the service itself
-    SERVICE_PACKAGES_FOR_SERVICE = {
-        'rabbitmq': 'rabbitmq-server',
-        'kafka': 'kafka-server'
-    }
-    # add this if the module name is not same as the service itself
-    ANSIBLE_MODULE_FOR_SERVICE = {
-        "redis": "redis-server",
-
-    }
+from abc import ABCMeta, abstractmethod, abstractproperty
+from collections import defaultdict, OrderedDict
+from itertools import groupby
 
-    def make_parser(self):
-        super(Service, self).make_parser()
-        self.parser.add_argument(
-            'service_group',
-            choices=self.SERVICES.keys(),
-            help="The service group to run the command on"
+import attr
+import six
+from clint.textui import puts, colored, indent
+
+from commcare_cloud.commands.ansible.helpers import AnsibleContext, get_django_webworker_name, \
+    get_formplayer_spring_instance_name, get_formplayer_instance_name, get_celery_workers
+from commcare_cloud.commands.ansible.run_module import run_ansible_module
+from commcare_cloud.commands.command_base import CommandBase
+from commcare_cloud.environment.main import get_environment
+from commcare_cloud.fab.exceptions import NoHostsMatch
+
+ACTIONS = ['start', 'stop', 'restart', 'status', 'help']
+
+STATES = {
+    'start': 'started',
+    'stop': 'stopped',
+    'restart': 'restarted',
+    'status': 'status'
+}
+
+
+@attr.s
+class ServiceOption(object):
+    name = attr.ib()
+    sub_options = attr.ib(default=None)
+
+
+class ServiceBase(six.with_metaclass(ABCMeta)):
+    @abstractproperty
+    def name(self):
+        raise NotImplementedError
+
+    @abstractproperty
+    def inventory_groups(self):
+        raise NotImplementedError
+
+    def __init__(self, environment, ansible_context):
+        self.environment = environment
+        self.ansible_context = ansible_context
+
+    def run(self, action, host_pattern=None, process_pattern=None):
+        if action == 'help':
```

Comment:
Nice add. Does this make `service` command behave differently than others? 
Would be nice if all commands keep the same interface.

Reply:
slightly, you still get the standard help if you use `-h` but this gives you more based on the env and options you've selected which I don't think argparser can do.

Reply:
I did some experiments on getting argparser to manage this help and it can be done but it's not pretty and doesn't output very well.

Reply:
There are certainly ways to change the help text, with only minor hacks 😉. Here are some examples of where we already do this
- https://github.com/dimagi/commcare-cloud/blob/abf7a894f8d1ae0ff53f4eb6f183159783ad81cd/src/commcare_cloud/commands/fab.py#L16
- https://github.com/dimagi/commcare-cloud/blob/abf7a894f8d1ae0ff53f4eb6f183159783ad81cd/src/commcare_cloud/commands/ansible/ansible_playbook.py#L38

You could probably just use the second method (with `add_to_help_text`) to append what's now printed with `help` to the the default help text. Maybe that's what you already figured out, and I agree making that pretty is probably an unbounded effort